server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MessageAppender.java

```diff
@@ -92,9 +92,11 @@ public MessageAppender(MailboxManager mailboxManager, AttachmentManager attachme
 
     private Flags getFlags(CreationMessage message) {
         return message.getOldKeyword()
-                .map(OldKeyword::asFlags)
-                .orElseGet(() -> keywordsOrDefault(message)
-                                    .asFlags());
+            .map(OldKeyword::asKeywords)
+            .map(Optional::of)
```

Comment:
Hum I think line 96 and 98 are useless no ?
For me at line 98 you will always have Optional.of(SOMETHING) because it is the result of line 96...
Oh I can see why you did that.
I think it's hard to understand what you did.
Is it more what you want to do:
```
.map(OldKeyword::asKeywords)
.orElse(message.getKeyWords().orElse(Keywords.DEFAULT_VALUE))
.asFlags();
```

Reply:
Latest versions is waaaaay simpler :trollface: 

```
return message.getKeywords().asFlags();
```

I can not do simpler sorry. What you commented on was an intermediary state... I try to be `pedagogic` and avoid `big diffs` as the team have PB handling them.